presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -139,6 +139,9 @@ public void testScalarSubQueryException()
             throws Exception
     {
         assertFails(NOT_SUPPORTED, "SELECT 'a', (VALUES (1)) GROUP BY 1");
+        assertFails(NOT_SUPPORTED, "SELECT 'a', (SELECT (1))");
+        analyze("SELECT * FROM (SELECT 1) t1(x) JOIN (SELECT 1) t2(x) ON t1.x=t2.x");
```

Comment:
This query does not really contain a scalar subquery, so I'd remove it from the test.